tests/framework/unit_tests/TSA/testRWD.py

```diff
@@ -0,0 +1,330 @@
+  xml.append(xmlUtils.newNode('FeatureIndex', text=f'{FeatureIndex}'))
+  xml.append(xmlUtils.newNode('SampleType', text=f'{SampleType}'))
+  return xml
+
+def createFromXML(xml):
+  print(vars(RWD))
+  rwd = RWD()
+  
+  inputSpec = rwd.getInputSpecification()()
+  inputSpec.parseNode(xml)
+  rwd.handleInput(inputSpec)
+  return rwd
+
+def createRWD(targets, SignatureWindowLength, FeatureIndex, SampleType):
+  xml = createRWDXML(targets, SignatureWindowLength, FeatureIndex, SampleType)
+  print('createRWDXML passed')
+  rwd = createFromXML(xml)
+  print('createFromXML passed')
+  return rwd
+
+'''
+def createRWDSignal(time_series, pivot, intercept=0, plot=False):
+  if plot:
+    import matplotlib.pyplot as plt
+    fig, ax = plt.subplots()
+  signal = np.zeros(len(pivot)) + intercept
+
+  # moving average: random noise lag
+  for q, theta in enumerate(noise):
+    signal[q+1:] += theta * noise[:-(q+1)]
+  # autoregressive: signal lag
+  for t, time in enumerate(pivot):
+    for p, phi in enumerate(time_series):
+      if t > p:
+        signal[t] += phi * signal[t - p - 1]
+  if plot:
+
+    ax.plot(pivot, signal, 'g.-')
+    plt.show()
+  return signal
+'''
+
+###################
+#  Simple         #
+###################
+# generate signal
+targets = ['A'] #, 'B', 'C']
+pivot = np.linspace(0, 100, 1000)
+N = len(pivot) 
+s = np.linspace(-10,10,1000)
+time_series = 3.5*s+s**2+10
+#noise = rng.random((1000,))*0.02
+#signalA, noise = createRWDSignal(time_series, noise, pivot, plot=plot)
+
+signals = np.zeros((len(pivot), 1))
+signals[:, 0] = time_series
+
+##########
+# Simplest reasonable case
+#
+rwd = createRWD(targets, 105, 3,0)
+settings = {'SignatureWindowLength':105, 'FeatureIndex': 3, 'SampleType' : 0}
+if 'SignatureWindowLength' not in settings:
+  print('SignatureWindowLength not in settings')
+settings = rwd.setDefaults(settings)
+params = rwd.characterize(signals, pivot, targets, settings)
+check = params['A']#['rwd']
+print('check', check.keys())
+
+
+# code to generate answer
+SignatureWindowLength = 105
+history = np.copy(time_series)
+sampleLimit = len(history)-SignatureWindowLength
+WindowNumber = sampleLimit//4
+#print(sampleLimit)
+print('WindowNumber',WindowNumber)
+sampleIndex = np.random.randint(sampleLimit, size=WindowNumber)
+#print('sampleIndex', sampleIndex)
+BaseMatrix = np.zeros((SignatureWindowLength, WindowNumber))
+for i in range(WindowNumber):
+  WindowIndex = sampleIndex[i]
+  BaseMatrix[:,i] = np.copy(history[WindowIndex:WindowIndex+SignatureWindowLength])
+
+AllWindowNumber = len(history)-SignatureWindowLength+1
+SignatureMatrix = np.zeros((SignatureWindowLength, AllWindowNumber))
+
+for i in range(AllWindowNumber):
+  SignatureMatrix[:,i] = np.copy(history[i:i+SignatureWindowLength])
+
+BaseMatrix = np.copy(SignatureMatrix)
+U,S,V = LA.svd(BaseMatrix,full_matrices=False)
+assert np.allclose(BaseMatrix, np.dot(U, np.dot(np.diag(S), V)))
+
+
+F = U.T @ SignatureMatrix  
```

Comment:
Please check https://github.com/idaholab/raven/wiki/RAVEN-Software-Coding-Standard and fix all violations